.github/workflows/license.yml

```diff
@@ -0,0 +1,81 @@
+name: Graph Compiler License Check
+
+on:
+  pull_request:
+
+permissions: read-all
+
+jobs:
+  license-check:
+    name: License Check
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+      repository-projects: write
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          submodules: true
+
+      - name: Get merge base
+        run: |
+          echo "MERGE_BASE=`git merge-base ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }}`" >> $GITHUB_ENV
+
+      - name: Get changed files
+        run: |
+          echo "CHANGED_FILES=`git diff --name-only $MERGE_BASE ${{ github.event.pull_request.head.sha }} | paste -sd,`" >> $GITHUB_ENV
+
+      - name: Perform license check
+        shell: python
+        run: |
+            import os, datetime, sys
+
+            intel_license: list[str] = [
```

Comment:
shall we put this script in `.github` or somewhere inside our repo?

Reply:
@Menooker 
If developers need to run it as a local check, it is ok to split the script into a seperate python file. 
Do you really need to do the local check?

Reply:
Please also note that we already have `script/` directory for compiling the repo and compiling LLVM. The `sh` files are used in CI too. I think we should add the py file to this directory.

Reply:
How about put this check script into `repo_dir/scripts/license_check.py`, so that this script can be used not only in actions but also from local.